contracts/tatum/Tatum721Provenance.sol

```diff
@@ -0,0 +1,223 @@
+//SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+pragma experimental ABIEncoderV2;
+
+import "../token/ERC721/extensions/ERC721Enumerable.sol";
+import "../token/ERC721/extensions/ERC721URIStorage.sol";
+import "../access/AccessControlEnumerable.sol";
+
+contract Tatum721Provenance is
+    ERC721Enumerable,
+    ERC721URIStorage,
+    AccessControlEnumerable
+{
+    
+    bytes32 public constant MINTER_ROLE = keccak256("MINTER_ROLE");
+    mapping(uint256 => string[]) private _tokenData;
+    mapping(uint256 => address[]) private _cashbackRecipients;
+    mapping(uint256 => uint256[]) private _cashbackValues;
+    mapping(uint256 => uint256[]) private _fixedValues;
+
+    event TransferWithProvenance(
+        uint256 indexed id,
+        address owner,
+        string data
+    );
+
+    constructor(string memory name_, string memory symbol_)
+        ERC721(name_, symbol_)
+    {
+        _setupRole(DEFAULT_ADMIN_ROLE, _msgSender());
+        _setupRole(MINTER_ROLE, _msgSender());
+    }
+
+    function _appendTokenData(
+        uint256 tokenId,
+        string memory tokenData
+    ) internal virtual {
+        require(
+            _exists(tokenId),
+            "ERC721URIStorage: URI set of nonexistent token"
+        );
+        _tokenData[tokenId].push(tokenData);
+    }
+
+    function mintWithCashback(
+        address to,
+        uint256 tokenId,
+        string memory uri,
+        address[] memory recipientAddresses,
+        uint256[] memory cashbackValues,
+        uint256[] memory fValues
+    ) public returns (bool) {
+        require(
+            hasRole(MINTER_ROLE, _msgSender()),
+            "ERC721PresetMinterPauserAutoId: must have minter role to mint"
+        );
+        _mint(to, tokenId);
+        _setTokenURI(tokenId, uri);
+        // saving cashback addresses and values
+        _cashbackRecipients[tokenId] = recipientAddresses;
+        _cashbackValues[tokenId] = cashbackValues;
+        _fixedValues[tokenId]=fValues;
+        return true;
+    }
+
+    function mintWithTokenURI(
```

Comment:
cant this function be replaced for mintMultiple and this logic mint without cashback put inside the mintWithCashback, where cashback will be used only if cashbackValues.length > 0

Reply:
I did this but as soon as I add mint multiple the bytecode size exceeds